apps/src/lab2/levelEditors/aichatSettings/EditAichatSettings.tsx

```diff
@@ -27,6 +30,50 @@ import ModelSelectionFields from './ModelSelectionFields';
 import VisibilityDropdown from './VisibilityDropdown';
 import {UpdateContext} from './UpdateContext';
 import moduleStyles from './edit-aichat-settings.module.scss';
+import {getTypedKeys} from '@cdo/apps/types/utils';
+
+function sanitizeSettings(settings: LevelAichatSettings) {
```

Comment:
Added a function to sanitize the settings whenever they're saved in levelbuilder. This will make sure we remove any unrecognized keys, and set any missing values to defaults. Useful when the properties structure changes after levels have already been created.